spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerMutationFactoryImpl.java

```diff
@@ -114,13 +118,49 @@ public Mutation delete(Class entityClass, Key key) {
 		return delete(entityClass, KeySet.singleKey(key));
 	}
 
-	private Mutation saveObject(Op op, Object object, Set<String> includeColumns) {
+	private List<Mutation> saveObject(Op op, Object object, Set<String> includeColumns) {
 		SpannerPersistentEntity<?> persistentEntity = this.spannerMappingContext
 				.getPersistentEntity(object.getClass());
+		List<Mutation> mutations = new ArrayList<>();
 		Mutation.WriteBuilder writeBuilder = writeBuilder(op,
 				persistentEntity.tableName());
 		this.spannerEntityProcessor.write(object, writeBuilder::set, includeColumns);
-		return writeBuilder.build();
+		mutations.add(writeBuilder.build());
+
+		persistentEntity.doWithChildCollectionProperties(spannerPersistentProperty -> {
+			Iterable kids = (Iterable) persistentEntity.getPropertyAccessor(object)
+					.getProperty(spannerPersistentProperty);
+			if (kids != null) {
+				for (Object child : kids) {
+					verifyChildHasParentId(persistentEntity, object,
+							this.spannerMappingContext.getPersistentEntity(
+									spannerPersistentProperty.getColumnInnerType()), child);
+					mutations.addAll(saveObject(op, child, includeColumns));
+				}
+			}
+		});
+		return mutations;
+	}
+
+	private void verifyChildHasParentId(SpannerPersistentEntity parentEntity,
```

Comment:
this change is a part of #912, should it be here as well?

Reply:
This one is different.

This one ensures that the values of the PKs are the same, so that one child doesn't get saved into some other parent. 

The other one is ensuring that the child POJO has the necessary properties.